packages/happn-db-provider-loki/index.js

```diff
@@ -607,14 +607,18 @@ module.exports = class LokiDataProvider extends commons.BaseDataProvider {
 
   copyTempDataToMain(callback) {
     if (fs.existsSync(this.settings.filename)) {
-      fs.unlinkSync(this.settings.filename);
+      fs.unlink(this.settings.filename, (err) => {
+        if (err) this.logger.error(err.message);
+      });
     }
-    fs.copy(this.settings.tempDataFilename, this.settings.filename, callback);
+    fs.rename(this.settings.tempDataFilename, this.settings.filename, callback);
   }
 
   copyMainDataToArchive(suffix, callback) {
     if (fs.existsSync(`${this.settings.filename}.${suffix}`))
-      fs.unlinkSync(`${this.settings.filename}.${suffix}`);
+      fs.unlink(`${this.settings.filename}.${suffix}`, (err) => {
+        if (err) this.logger.error(err.message);
+      });
     fs.copy(this.settings.filename, `${this.settings.filename}.${suffix}`, callback);
```

Comment:
this looks suspect - you're not waiting for unlink to complete (if it needs to) before copying the file, so we have a race condition. Best would be to have a callback-based function that verifies that the old file does not exist (by deleting if necessary), and only starting the copy in its callback.

Reply:
unlinkSync is a synchronous operation, copy will never be called during this operation.